src/private/files.ts

```diff
@@ -226,6 +226,43 @@ export namespace files {
     listUrl?: string;
   }
 
+  /**
+   * @private
+   * Hide from docs
+   *
+   * Download Files interface
+   */
+  export interface IFileItem {
+    /**
+     * ID of the file metadata
+     */
+    objectId?: string;
+    /**
+     * Path of the file
+     */
+    path?: string;
+    /**
+     * Size of the file
+     */
+    size?: number;
+    /**
+     * Download status
+     */
+    status?: string;
+    /**
+     * Download timestamp
+     */
+    timestamp: Date | string;
+    /**
+     * File name
+     */
+    title: string;
+    /**
+     * Type of file
+     */
+    type: string;
```

Comment:
>type: string; [](http://example.com/codeflow?start=4&length=13)

What does type actually indicate or contain? Is this file extension? Could it be represented as a limited set of possible values in an enum or type? I realize this is a microsoft internal function but it will be very hard for other microsoft developers to use without more information.

Reply:
As you rightly guessed, type indicates file extension here. Updated the doc comment for the same. The reason it is represented as string and not type is to avoid the complexity involved in doing so because of huge list of such extensions and/or avoid missing any extension from the type list which might be encountered dynamically. The respective client app code is handling it to infer required icon details etc. Note that the underlying client API output data structure represents this field as string too.

Reply:
If this is an extension, the property should be called `extension`.

Reply:
Since the property is named as **type** in Teams Web Client, hence retained the same name here, so that the conversion/switch can happen seamlessly between SDK and Teams Web Client code.
@ydogandjiev Please correct me if it can work otherwise, as suggested by Marcin.

Reply:
As discussed offline, this is not good enough of a reason for me. We need to focus on designing the best possible developer experience for the developers using the SDK. Not guide the design based on implementation details of Teams (or any other host) trying to host these.

Reply:
Suggestion incorporated.